### DISCUSSION TITLE
Tailwind 3 not generating css

### DISCUSSION THREAD

**Discussion by Pratitb** [tailwindlabs/tailwindcss#7222] [Help]
I upgraded from tailwind 2 to 3. 
Now whenever I run "npm build" it shows "No utility classes were detected in your source files. If this is unexpected, double-check the `content` option in your Tailwind CSS configuration."
CSS classes are not being generated in my output file. And how do I get the just-in-time engine to generate css for my existing html code.

**Comment by adamwathan** [comment]
Hey! You need configure the `content` option in your Tailwind config file, which is what the error message is trying to tell you to do by including a link to that section in the documentation.

https://tailwindcss.com/docs/content-configuration

Is there anything we can do better with that error message that would have gotten you to actually click that link and follow the instructions there?

**Reply by Pratitb** [reply]
Can you please help me with what to change and what do I write for my project. I'll share some screenshots for your reference

![image](https://user-images.githubusercontent.com/26714578/151304618-2ddc8074-aa40-4f7c-af1b-53de0c73b51d.png)

![image](https://user-images.githubusercontent.com/26714578/151304728-f1da695f-7a9c-443e-872a-81fbc1cd5b60.png)

**Reply by luiscardona88** [reply]
I had the same issue and previous comment say the correct way  when you compile  for example like this 

npx tailwindcss -i **input.css** -o **output.css** --watch      both css need to have in  the same level(same directoy and all your output files too html, css etc)

**Reply by ddomdel** [reply]
I'm having the same issue as well 
It only generates the base, and it doesn't generate the components and utilities
![image](https://user-images.githubusercontent.com/78736225/153837770-4014ca40-ba3a-4add-b4e1-14eec742fa13.png)

![image](https://user-images.githubusercontent.com/78736225/153837712-29f57f31-6226-4945-81d5-76c08235d586.png)

![image](https://user-images.githubusercontent.com/78736225/153837346-10fb2c9f-bc0a-44b5-b2f5-b2ce8c185e88.png)
![image](https://user-images.githubusercontent.com/78736225/153837394-44c7492c-931b-48f5-b9aa-090678456cdf.png)
![image](https://user-images.githubusercontent.com/78736225/153837485-a7504551-f502-40f8-b305-96d8faa697fc.png)

**Comment by pm-patel12** [comment]
Same problem is here , try all the stuff to solve this also try lots of things from internet but there is same problem again plz , help
![Screenshot (198)](https://user-images.githubusercontent.com/83482652/154274195-3b32de99-6077-4b89-91b9-42637332b857.png)
![Screenshot (199)](https://user-images.githubusercontent.com/83482652/154274203-175a0c72-0006-4faa-9c42-f55b59a261e4.png)
![Screenshot (200)](https://user-images.githubusercontent.com/83482652/154274206-19793d3a-cecb-44b5-9353-b36b1dae2f4d.png)
![Screenshot (201)](https://user-images.githubusercontent.com/83482652/154274211-3b289beb-2fd1-4c74-bba5-25848fb25069.png)
![Screenshot (202)](https://user-images.githubusercontent.com/83482652/154274214-ab282be9-2a7f-4212-a3a2-c1c7f0e16966.png)
 🤞

**Comment by luiscardona88** [comment]
I solved this issue following these instructions:

1) npm init

2) npm install

3) npx tailwindcss init

4) npm install -D tailwindcss postcss autoprefixer

5) create input.css add css directives
![image](https://user-images.githubusercontent.com/32369325/154364806-62a2de13-3d55-466e-8b47-9d6e1fc7edd9.png)

@tailwind base;
@tailwind components;
@tailwind utilities

6) create empty output.css 

7) create index.html and add output.css link

![image](https://user-images.githubusercontent.com/32369325/154364875-fefb2255-f369-4269-8b79-c1abd8891079.png)


8) before compile ensure in my case I have to add each component like these in tailwind.config.js
![image](https://user-images.githubusercontent.com/32369325/154367449-d42c3bf3-00d2-4fe6-a8ec-9b75b5980e00.png)


![image](https://user-images.githubusercontent.com/32369325/154365013-a5ea376a-f90a-4689-a39b-6eec48ff8c22.png)


9) and then ensure you have the input.css and output.css and also html (all your files) in the same directory level

![image](https://user-images.githubusercontent.com/32369325/154365042-830f2b50-9014-45cc-8354-07d54c558c6b.png)

10) finally compile with  npx tailwindcss -i input.css -o output.css --watch

![image](https://user-images.githubusercontent.com/32369325/154365451-b47e734f-5a56-474c-9dc9-d5236d766300.png)

**Comment by dgtlmonk** [comment]
For those who are working with Vite + Svelte (as `module` type in package.json) and using `tailwind.config.cjs`. Make sure to specify your config file in the cli command

`tailwindcss -i in.css -o out.css -c tailwind.config.cjs`

**Reply by lblds** [reply]
its "-c tailwind.config.cjs" .. thanks
